src/main/java/htsjdk/samtools/util/BufferedLineReader.java

```diff
@@ -41,6 +41,25 @@
  */
 public class BufferedLineReader extends LineNumberReader implements LineReader {
 
+    private static class StringBackedInputStream extends InputStream {
+        private int idx = 0;
+        private final String str;
+        private final int len;
+
+        StringBackedInputStream(String str) {
+            this.str = str;
+            this.len = str.length();
+        }
+
+        @Override
+        public int read() throws IOException {
+            if(idx >= len) {
+                return -1;
+            }
+            return (int) str.charAt(idx++);
```

Comment:
@lbergelson I think solves the previous issue, where we cast as `byte` instead of `int`